### PR TITLE
DevTest: Add nodes and items for testing overlays

### LINKED ISSUE #12286: No `overlay_tiles` test nodes in devtest
https://github.com/minetest/minetest/blob/87472150bcb83e9cbad2f567ac536de0456ceb70/doc/lua_api.txt#L809-L850

### PR DESCRIPTION
**Pull request by Wuzzy2**
Fixes #12286.

For DevTest, tests for `overlay_tiles`, `inventory_overlay` and `wield_overlay` are added. Various combinations with the "global" `color` argument in the item definition are added as well. The nodes have one base tile and one overlay tile each. The craftitems were added to test the color set by item meta.

The tooltips describe what each item does.

## How to test

Read the code to see the item names, then give yourself these items in DevTest. Place them on the ground. Look at them in the inventory. Wield them. For the craftitems, one of them can be punched/placed to set a random color in item meta.

## Bug discovered during creation of this PR

* Wield image of the 2 craftitems remains uncolorized

### CODE REVIEW

File: games/devtest/mods/testitems/init.lua [hunk 1]
```diff
@@ -0,0 +1,55 @@
+local S = minetest.get_translator("testitems")
```
Comment (sfan5) [inline games/devtest/mods/testitems/init.lua:1]: devtest doesn't use translations

**Comment by Wuzzy2** [comment]
> devtest doesn't use translations

Other files in `testnodes` have this, too.

**Comment by Wuzzy2** [comment]
Note DevTest uses the `S()` symbol in a lot of places, actually. But if you propose to just nuke it all (except for the explicit translation testing), feel free to request it from me, I probably make a PR just for that.

**Comment by x2048** [comment]
@Wuzzy2  Coloring of wield_image should be fixed in 8756b7a735084c6a80f5ab031fa1d889528d6fde

**Review by SmallJoker — APPROVED** [review]
Works.

**Review by sfan5 — APPROVED** [review]
Works.
I hope someone adds search to the chest because for every devtest PR I review 3 minutes are wasted looking for the right items.

**Comment by SmallJoker** [comment]
unified_inventory and i3 are pretty practical for testing. I suggest to use an additional inventory mod to not deal with the currently rather impractical "Everything chest".

**Pull request merged**